MacDown/Code/Preferences/MPTerminalPreferencesViewController.m

```diff
@@ -0,0 +1,226 @@
+}
+
+- (NSImage *)toolbarItemImage
+{
+    return [NSImage imageNamed:@"PreferencesTerminal"];
+}
+
+- (NSString *)toolbarItemLabel
+{
+    return NSLocalizedString(@"Terminal", @"Preference pane title.");
+}
+
+#pragma mark - Private methods
+
+/**
+ * Searches for the the macdown shell utility and invokes foundShellUtilityAtURL: if found.
+ */
+- (void)lookForShellUtility
+{
+    NSTask *brewPrefixTask = [NSTask new];
+    [brewPrefixTask setLaunchPath:@"brew"];
+    [brewPrefixTask setArguments:@[@"--prefix"]];
+    self->brewPrefixOutputPipe = [NSPipe pipe];
+    [brewPrefixTask setStandardOutput:self->brewPrefixOutputPipe];
+    [[NSNotificationCenter defaultCenter] addObserver:self
+        selector:@selector(brewPrefixReadCompleted:)
+        name:NSFileHandleReadToEndOfFileCompletionNotification
+        object:[self->brewPrefixOutputPipe fileHandleForReading]];
+    [[self->brewPrefixOutputPipe fileHandleForReading] readToEndOfFileInBackgroundAndNotify];
+    
+    @try {
+        [brewPrefixTask launch];
+    }
+    @catch (NSException *exception) { // Homebrew not installed
+        if ([exception.name isEqualToString:NSInvalidArgumentException])
+        {
+            // If installed through DMG the macdown binary should be here
+            NSString *shellUtilityDefaultPath = @"/usr/local/bin/macdown";
+            
+            if ([[NSFileManager defaultManager] fileExistsAtPath:shellUtilityDefaultPath])
+            {
+                NSURL *shellUtilityUrl = [NSURL fileURLWithPath:shellUtilityDefaultPath];
+                [self foundShellUtilityAtURL:shellUtilityUrl];
+            }
+        }
+    }
+}
+
+- (void)brewPrefixReadCompleted:(NSNotification *)notification
+{
+    [[NSNotificationCenter defaultCenter] removeObserver:self
+        name:NSFileHandleReadToEndOfFileCompletionNotification
+        object:[notification object]];
+    
+    if ([notification object])
+    {
+        NSFileHandle *fileHandle = [notification object];
+        
+        if (fileHandle == self->brewPrefixOutputPipe.fileHandleForReading)
+        {
+            NSString *output = [[NSString alloc] initWithData:
+                                [[notification userInfo] objectForKey:NSFileHandleNotificationDataItem]
+                                encoding:NSUTF8StringEncoding];
+            output = [output stringByTrimmingCharactersInSet:
+                      [NSCharacterSet whitespaceAndNewlineCharacterSet]];
+            
+            NSString *shellUtilityPath = [output stringByAppendingString:@"/bin/macdown"];
+            
+            // If MacDown was installed from the Homebrew cask then the macdown binary should exist at this path
+            if ([[NSFileManager defaultManager] fileExistsAtPath:shellUtilityPath])
+            {
+                NSURL *shellUtilityUrl = [NSURL fileURLWithPath:shellUtilityPath];
+                [self foundShellUtilityAtURL:shellUtilityUrl];
+            }
+        }
+    }
+}
```

Comment:
I don’t really like this part. In general it would work, but holding only one pipe at a time as an instance variable means this method is non-reentrant.

I would extract this logic (including the notification handler) into its own controller. The controller can store a collection of pipes as they are instantiate, and clean them up in the handler. This is a bit difficult to explain…